packages/phone-number-privacy/signer/src/signing/get-partial-signature.ts

```diff
@@ -5,73 +5,121 @@ import {
   hasValidQueryPhoneNumberParam,
   hasValidTimestamp,
   isBodyReasonablySized,
-  logger,
   phoneNumberHashIsValidIfExists,
   SignMessageResponse,
   SignMessageResponseFailure,
   WarningMessage,
 } from '@celo/phone-number-privacy-common'
+import Logger from 'bunyan'
 import { Request, Response } from 'express'
+import allSettled from 'promise.allsettled'
 import { computeBlindedSignature } from '../bls/bls-cryptography-client'
 import { respondWithError } from '../common/error-utils'
+import { Counters, Histograms, Labels } from '../common/metrics'
 import { getVersion } from '../config'
 import { incrementQueryCount } from '../database/wrappers/account'
 import { getRequestExists, storeRequest } from '../database/wrappers/request'
 import { getKeyProvider } from '../key-management/key-provider'
+import { Endpoints } from '../server'
 import { getBlockNumber, getContractKit } from '../web3/contracts'
 import { getRemainingQueryCount } from './query-quota'
 
+allSettled.shim()
+
 export interface GetBlindedMessagePartialSigRequest {
   account: string
   blindedQueryPhoneNumber: string
   hashedPhoneNumber?: string
   timestamp?: number
+  sessionID?: string
 }
 
 export async function handleGetBlindedMessagePartialSig(
   request: Request<{}, {}, GetBlindedMessagePartialSigRequest>,
   response: Response
 ) {
-  logger.info('Begin handleGetBlindedMessagePartialSig request')
+  Counters.requests.labels(Endpoints.GET_BLINDED_MESSAGE_PARTIAL_SIG).inc()
+
+  const logger: Logger = response.locals.logger
+  logger.debug({ req: request.body }, 'Request received')
+  if (!request.body.sessionID) {
+    logger.warn({ req: request.body }, 'Request does not have sessionID')
+  }
+  logger.info('Begin handleGetBlindedMessagePartialSig')
+
+  if (!request.body.sessionID) {
```

Comment:
Why not combine this with the conditional at line 45?

Reply:
Good catch